### PR TITLE
Deprecate skipna=None for mad

### PR DESCRIPTION
**Pull request by phofl**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -10600,6 +10600,13 @@ def mad(self, axis=None, skipna=None, level=None):
         {examples}
         """
         if skipna is None:
+            warnings.warn(
+                "Passing None for skipna is deprecated and will raise in a future"
+                "version. Pass True instead. Only boolean values will be allowed "
```
Comment (mroeschke) [inline pandas/core/generic.py:10605]: Maybe good to add a separate `if not is_bool` to warn on non boolean values so it's easier to enforce in the future.

Comment (phofl) [reply pandas/core/generic.py:10605]: Done

**Comment by jreback** [comment]
hmm are any of the doc-string failures related here? (don't think so but hard to tell at a glance).

also in future, feel free to link to the deprecation removal issue when appropriate

**Comment by phofl** [comment]
Could not find anything which is related. Will link to the issue in the future

**Comment by jreback** [comment]
oh sorry what meant was add an entry in the deprecation removal issue (we have 1 master one)

i did for this one already - but FYI

**Comment by phofl** [comment]
Sorry my answer was very misleading. I meant:

I could not find any failing doctests which are related to this pr.
Will link to the deprecation issue in the future.


Sorry for the confusion

File: pandas/core/generic.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/generic.py:10605]: Nit: Could you use `pandas.core.dtypes.common.is_bool`

Comment (phofl) [reply pandas/core/generic.py:10605]: Done

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
can you rebase and ping on green

**Comment by phofl** [comment]
ping @jrbeack

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**